internal/pkg/workspace/workspace.go

```diff
@@ -489,3 +482,27 @@ func RelPath(fullPath string) (string, error) {
 	}
 	return path, nil
 }
+
+// WorkloadManifest represents raw local workload manifest.
+type WorkloadManifest []byte
+
+func (w WorkloadManifest) workloadName() (string, error) {
+	wl := struct {
+		Name string `yaml:"name"`
+	}{}
+	if err := yaml.Unmarshal(w, &wl); err != nil {
+		return "", err
```

Comment:
nit: should we return a wrapped error here? 
```suggestion
		return "", fmt.Errorf("unmarshal manifest file to retrieve 'name': %w", err)
```